src/cata_tiles.cpp

```diff
@@ -680,43 +716,72 @@ tile_type &cata_tiles::load_tile(JsonObject &entry, const std::string &id, int o
 {
     tile_type &curr_subtile = tile_ids[id];
 
-    if ( entry.has_array("fg") ) {
-        JsonArray fg_array = entry.get_array("fg");
-        while (fg_array.has_more()) {
-            const int fg = fg_array.next_int();
-            if ( fg >= 0 ) {
-                curr_subtile.fg.push_back(fg);
+    // loop through fg and bg
+    for (auto g : {std::make_pair("fg",&curr_subtile.fg), std::make_pair("bg",&curr_subtile.bg)}) {
+        // one sprite_variation for bookkeeping purposes
+        g.second->push_back(sprite_variation());
+        g.second->at(0).weight = -1;
+        // json array indicates rotations or variations
+        if ( entry.has_array(g.first) ) {
+            JsonArray g_array = entry.get_array(g.first);
+            // int elements of array indicates rotations
+            // create one variation, populate sprite_ids with list of ints
+            if ( g_array.test_int() ) {
+                g.second->push_back(sprite_variation());
+                while (g_array.has_more()) {
+                    const int s = g_array.next_int();
+                    if ( s >= 0 ) {
+                        g.second->at(0).sprite_ids.push_back(s);
```

Comment:
good argument in favor of function usage, or at least a named reference.

also, there's probably a bug here, related to which fg/bg element to write to. probably caused by me being confused by the deep nesting you're complaining about here.